document-store/src/main/java/org/hypertrace/core/documentstore/expression/UnaryOperatorExpression.java

```diff
@@ -0,0 +1,18 @@
+package org.hypertrace.core.documentstore.expression;
+
+import lombok.AllArgsConstructor;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+
+@AllArgsConstructor
+@Getter
+@EqualsAndHashCode
+public class UnaryOperatorExpression implements OperatorExpression {
```

Comment:
Right! We could easily generalise it. Will change the implementation.

Reply:
Some of this support was already in the existing implementation right? Are we re-implementing it because it wasn't strongly typed, isn't usable here or some other shortcoming?

Reply:
This is for implementing GroupBy.